Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/ReflectometryILLPreprocess.py

```diff
@@ -559,7 +545,13 @@ def _waterCalibration(self, ws):
         if self.getProperty(Prop.WATER_REFERENCE).isDefault:
             return ws
         waterWS = self.getProperty(Prop.WATER_REFERENCE).value
-        # input validation for InputWorkspace compatibility, but runs?
+        detWSName = self._names.withSuffix('water_detectors')
+        ExtractMonitors(InputWorkspace=waterWS,
+                        DetectorWorkspace=detWSName,
+                        EnableLogging=self._subalgLogging)
+        if mtd.doesExist(detWSName) is None:
+            raise RuntimeError('No detectors in the water reference data.')
+        waterWS = mtd[detWSName]
```

Comment:
We should perhaps try to avoid `mtd`. Please use this style instead:
```python
waterWS = ExtractMonitors(
    InputWorkspace=waterWS,
    DetectorWorkspace=detWSName,
    EnableLogging=self._subalgLogging)
```
In the future, if we start using `StoreInADS=False`, the above style is easier to convert from.